### PR TITLE
Add Client tests

### PR DESCRIPTION
**Pull request by mrliptontea**
* Add simple tests for the Client class, which performs HTTP requests, to ensure we send them correctly.
* Disable PHP 8+ tests as current version of Guzzle doesn't work there.
* Add httpbin to CircleCI config.
* Format and update README.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,6 +19,10 @@ jobs:
         type: string
     docker:
       - image: cimg/php:<< parameters.php_version >>
+        environment:
+          HTTPBIN_HOST: http://httpbin
```
Comment (haakym) [inline .circleci/config.yml:23]: I don't fully understand this, but feel I must ask if this is this a typo?

```suggestion
          HTTPBIN_HOST: http://httpbin.org
```

Comment (haakym) [reply .circleci/config.yml:23]: Also wondering if it should be `https`. I see both are available as described here https://hub.docker.com/r/kennethreitz/httpbin so maybe it doesn't matter?

Comment (mrliptontea) [reply .circleci/config.yml:23]: This references the container below, so hits it locally.

https://github.com/talis/SRUClient-php/blob/8929a38b0ac702051ca0b6005317c5acc9cf72ed/.circleci/config.yml#L24-L25

**Review by rsinger — APPROVED** [review]
LGTM

**Pull request merged**